homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,644 @@
+        return []
+
+    port_forwards_count: int = resp["NewPortMappingNumberOfEntries"]
+
+    _LOGGER.debug(
+        "Specific %s response: GetPortMappingNumberOfEntries=%s",
+        SWITCH_TYPE_PORTFORWARD,
+        port_forwards_count,
+    )
+
+    for i in range(port_forwards_count):
+
+        portmap = service_call_action(
+            fritzbox_tools,
+            con_type,
+            "1",
+            "GetGenericPortMappingEntry",
+            NewPortMappingIndex=i,
+        )
+
+        if not portmap:
+            _LOGGER.debug("The FRITZ!Box has no %s options", SWITCH_TYPE_DEFLECTION)
+        else:
+            _LOGGER.debug(
+                "Specific %s response: GetGenericPortMappingEntry=%s",
+                SWITCH_TYPE_PORTFORWARD,
+                portmap,
+            )
+
+            # We can only handle port forwards of the given device
+            local_ip = async_get_source_ip(fritzbox_tools.host)
+            _LOGGER.debug("IP source for %s is %s", fritzbox_tools.host, local_ip)
+            if portmap["NewInternalClient"] == local_ip:
+                entities_list.append(
+                    FritzBoxPortSwitch(
+                        fritzbox_tools,
+                        device_friendly_name,
+                        portmap,
+                        i,
+                        con_type,
+                    )
+                )
+
+    return entities_list
+
+
+def profile_entities_list(
+    fritzbox_tools: FritzBoxTools, device_friendly_name: str
+) -> list[FritzBoxProfileSwitch]:
+    """Get list of profile entities."""
+    _LOGGER.debug("Setting up %s switches", SWITCH_TYPE_DEVICEPROFILE)
+    entities_list: list = []
+    if len(fritzbox_tools.fritz_profiles) > 0:
+        for profile in fritzbox_tools.fritz_profiles.keys():
+            entities_list.append(
+                FritzBoxProfileSwitch(fritzbox_tools, device_friendly_name, profile)
+            )
+    return entities_list
+
+
+def wifi_entities_list(
+    fritzbox_tools: FritzBoxTools, device_friendly_name: str
+) -> list[FritzBoxWifiSwitch]:
+    """Get list of wifi entities."""
+    _LOGGER.debug("Setting up %s switches", SWITCH_TYPE_WIFINETWORK)
+    entities_list: list = []
+    std_table = {"ac": "5Ghz", "n": "2.4Ghz"}
+    networks: dict = {}
+    for i in range(4):
+        if ("WLANConfiguration" + str(i)) in fritzbox_tools.connection.services:
+            network_info = service_call_action(
+                fritzbox_tools, "WLANConfiguration", str(i), "GetInfo"
+            )
+            if network_info:
+                ssid = network_info["NewSSID"]
+                if ssid in networks.values():
+                    networks[i] = f'{ssid} {std_table[network_info["NewStandard"]]}'
+                else:
+                    networks[i] = ssid
+
+    for net in networks:
+        entities_list.append(
+            FritzBoxWifiSwitch(fritzbox_tools, device_friendly_name, net, networks[net])
+        )
+
+    return entities_list
```

Comment:
You can use a list comp here

Reply:
Done. Thx for the explanation as I was misunderstanding the comment!